src/views/login.js

```diff
@@ -1,48 +1,79 @@
 import {Text, View, Button, StyleSheet, TextInput} from 'react-native';
-import React, {useState} from 'react';
+import React, {useContext, useState} from 'react';
+import AuthContext from '../store/authContext';
 
-const validateEmail = (email) => {
-  var re = /^(([^<>()\[\]\\.,;:\s@”]+(\.[^<>()\[\]\\.,;:\s@”]+)*)|(“.+”))@((\[[0–9]{1,3}\.[0–9]{1,3}\.[0–9]{1,3}\.[0–9]{1,3}])|(([a-zA-Z\-0–9]+\.)+[a-zA-Z]{2,}))$/;
-  return re.test(email);
-};
-
-export const Login = ({navigation}) => {
+export const Login = () => {
   const [email, onChangeEmail] = useState('');
   const [password, onChangePassword] = useState('');
+  const [isValidEmail, setIsValidEmail] = useState(true);
+  const [isValidPassword, setIsValidPassword] = useState(true);
+  const {handleIsLoggedIn} = useContext(AuthContext);
 
   return (
     <View style={styles.container}>
       <View style={styles.inputWrapper}>
         <Text>Email:</Text>
         <TextInput
-          style={styles.input}
+          style={isValidEmail ? styles.input : styles.input_error}
```

Comment:
Um truque que podes usar aqui
```suggestion
          style={[isValidEmail, (styles.input && styles.input_error)]}
```